File: app/src/main/java/com/nfcalarmclock/permission/NacRequiredPermissionPreference.java
package com.nfcalarmclock.permission;

import android.content.Context;
import android.content.res.Resources;
import android.util.AttributeSet;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.preference.Preference;
import androidx.preference.PreferenceViewHolder;
import com.nfcalarmclock.R;

/**
 * A preference that is used to display optional permissions.
 */
public class NacRequiredPermissionPreference
	extends Preference
{

	/**
	 */
	public NacRequiredPermissionPreference(Context context)
	{
		super(context);
		this.init();
	}

	/**
	 */
	public NacRequiredPermissionPreference(Context context, AttributeSet attrs)
	{
		super(context, attrs);
		this.init();
	}

	/**
	 */
	public NacRequiredPermissionPreference(Context context, AttributeSet attrs, int style)
	{
		super(context, attrs, style);
		this.init();
	}

	/**
	 * Initialize the preference.
	 */
	private void init()
	{
		setLayoutResource(R.layout.nac_preference_permission);
	}

	/**
	 */
	@Override
	public void onBindViewHolder(@NonNull PreferenceViewHolder holder)
	{
		super.onBindViewHolder(holder);

		Context context = getContext();
		Resources res = context.getResources();
		TextView permissionType = (TextView) holder.findViewById(R.id.permission_type);

		// Get the text and color
		String text = res.getString(R.string.message_required).toLowerCase();
		int color = res.getColor(R.color.red);

		// Setup the textview
		permissionType.setText(text);
		permissionType.setTextColor(color);
	}

}

File: app/src/main/kotlin/com/nfcalarmclock/system/permission/NacRequiredPermissionPreference.kt
package com.nfcalarmclock.system.permission

import android.content.Context
import android.util.AttributeSet
import android.widget.TextView
import androidx.preference.Preference
import androidx.preference.PreferenceViewHolder
import com.nfcalarmclock.R
import java.util.Locale

/**
 * A preference that is used to display optional permissions.
 */
class NacRequiredPermissionPreference @JvmOverloads constructor(

	/**
	 * Context.
	 */
	context: Context,

	/**
	 * Attribute set.
	 */
	attrs: AttributeSet? = null,

	/**
	 * Default style.
	 */
	style: Int = 0

	// Constructor
): Preference(context, attrs, style)
{

	/**
	 * Constructor.
	 */
	init
	{
		layoutResource = R.layout.nac_preference_permission
	}

	/**
	 * Called when the view holder is bound.
	 */
	override fun onBindViewHolder(holder: PreferenceViewHolder)
	{
		// Super
		super.onBindViewHolder(holder)

		// Get the text view
		val permissionType = holder.findViewById(R.id.permission_type) as TextView

		// Setup the textview
		setPermissionText(permissionType)
		setPermissionTextColor(permissionType)
	}

	/**
	 * Set the permission text.
	 */
	private fun setPermissionText(textView: TextView)
	{
		// Get the message
		val locale = Locale.getDefault()
		val message = context.resources.getString(R.string.message_required)

		// Set the text
		textView.text = message.lowercase(locale)
	}

	/**
	 * Set the color of the permission text.
	 */
	private fun setPermissionTextColor(textView: TextView)
	{
		// Get the color based on the API
		val color = context.getColor(R.color.red)

		// Set the color
		textView.setTextColor(color)
	}

}